node/index.d.ts

```diff
@@ -1891,8 +1891,27 @@ declare module "dns" {
         priority: number
     }
 
-    export function lookup(domain: string, family: number, callback: (err: Error, address: string, family: number) => void): string;
-    export function lookup(domain: string, callback: (err: Error, address: string, family: number) => void): string;
+    // Supported getaddrinfo flags
+    export const ADDRCONFIG: number;
+    export const V4MAPPED: number;
+
+    export interface LookupOptions {
+        family?: number;
+        hints?: number;
+        all?: boolean;
+    }
+
+    interface Address {
+        address: string;
+        family: number;
+    }
+
+    interface QueryReqWrap {}
+
+    export function lookup(domain: string, family: number, callback: (err: NodeJS.ErrnoException, address: string, family: number) => void): QueryReqWrap;
+    export function lookup(domain: string, options: LookupOptions, callback: (err: NodeJS.ErrnoException, address: string, family: number) => void): QueryReqWrap;
+    export function lookup(domain: string, options: LookupOptions, callback: (err: NodeJS.ErrnoException, addresses: Address[]) => void): QueryReqWrap;
```

Comment:
I think it would be possible to add a type
```
interface LookupAllOptions {
    family?: number;
    hints?: number;
    all: true;
}
```
and use it here instead `LookupOptions` to nail down the callback using `Address[]` works only in case `all` is set to true.